test/e2e/knative_serving_test.go

```diff
@@ -301,3 +304,58 @@ func waitForRouteServingText(t *testing.T, caCtx *test.Context, routeDomain, exp
 		t.Fatalf("The Route at domain %s didn't serve the expected text \"%s\": %v", routeDomain, expectedText, err)
 	}
 }
+
+func testKnativeServingForGlobalProxy(t *testing.T, caCtx *test.Context) {
+	t.Log("update global proxy with empty value")
+	if err := test.UpdateGlobalProxy(caCtx, ""); err != nil {
+		t.Fatal("Failed to update proxy", err)
+	}
+
+	t.Log("deploy successfully knative service after proxy update")
+	if _, err := test.WithServiceReady(caCtx, proxyHelloworldServiceSuccess, testNamespace, image); err != nil {
+		t.Fatal("Knative Service not ready", err)
+	}
+
+	t.Log("update global proxy with proxy server")
+	if err := test.UpdateGlobalProxy(caCtx, "http://1.2.4.5:8999"); err != nil {
+		t.Fatal("Failed to update proxy", err)
+	}
+
+	t.Log("wait for controller to be ready after update")
+	if err := test.WaitForControllerEnvironment(caCtx, knativeServing); err != nil {
+		t.Fatal(err)
+	}
+
+	t.Log("deploy knative service after proxy update")
+	test.WithServiceReady(caCtx, proxyHelloworldService, testNamespace, proxyImage)
+
+	svc, err := test.GetService(caCtx, proxyHelloworldService, testNamespace)
+	if err != nil {
+		t.Fatal(err)
+	}
+	var proxyExist bool
+	for _, cond := range svc.Status.Conditions {
+		// After global proxy update every call goes through proxy server
+		// Here it give unable to pull image because it tries to connect to not running http server
+		if strings.Contains(cond.Message, "dial tcp 1.2.4.5:8999: i/o timeout") {
+			// use bool variable here because service status have more than one conditions and if none of the condition matches
+			// then assume knative service failed because of some other reason
+			proxyExist = true
+			break
+		}
+	}
+	if !proxyExist {
+		t.Fatal("Service not ready", err)
+	}
+
+	t.Log("update global proxy with empty value")
+	if err = test.UpdateGlobalProxy(caCtx, ""); err != nil {
```

Comment:
We need to do this on failure too, right? I'd `defer` + `OnInterrupt` this before setting the proxy to make sure we clean up after us correctly in this test.